controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -815,6 +815,19 @@ private Throwable getRealException(Throwable ex) {
                 .thenApplyAsync(res -> TxnState.newBuilder().setState(TxnState.State.valueOf(res.name())).build(), executor);
     }
 
+    /**
+     * List txnIds in completed state.
+     * @param scope scope name
+     * @param stream stream name
+     * @param requestId request id
+     * @return list of txnIds future
+     */
+    public CompletableFuture<List<UUID>> listCompletedTxns(final String scope, final String stream, final long requestId) {
```

Comment:
`Exceptions.checkNotNullOrEmpty(scope, "scope");`
`Exceptions.checkNotNullOrEmpty(stream, "stream");`

Reply:
Applied above checks in method.